File: sources/2017-01-15-From-Java-to-Go/demo/Switch.java
public class Switch {
  public static void main(String ... args) {
    int i=2;
    // tag::main[]
    switch i {
    case 0:
      System.out.print("None");
      break;
    case 1:
      System.out.print("Single ")
    default:
      System.out.println("Thing")
    }
    // end::main[]
  }
}

File: sources/2017-01-15-From-Java-to-Go/demo/switch.go
package main

import (
	"fmt"
)

func patternMatch(i int) {
	// tag::pattern[]
	switch {
	case i == 0:
		fmt.Print("None")
	case i == 1:
		fmt.Print("One ")
	case i > 1:
		fmt.Print("Many ")
	default:
		fmt.Print("Invalid")
	}
	// end::pattern[]
}

func main() {
	i := 1
	// tag::main[]
	switch i {
	case 0:
		fmt.Print("None")
	case 1:
		fmt.Print("Single ")
		fallthrough
	default:
		fmt.Print("Thing")
	}
	// end::main[]
}
